### ISSUE TITLE
data_catalog.yml generated by DataCatalog.export_data() has mismatched paths

### ISSUE THREAD

**Issue by dumontgoulart** [Deltares/hydromt#230]
Hi,

When using the export_data() from DataCatalog.py, the relative path to the data inside the data_catalog.yml is not aligned with the actual location of the data_catalog.yml file. More specifically: the data_catalog.yml file is located inside the catalog, but its relative path calls the root file again before the files location (calling the root twice). 
 To reproduce:
```
import hydromt

hydromt.__version__   ### '0.5.0' Edit: also happens in '0.6.0'

# Open catalog:
data_cat = hydromt.DataCatalog(data_libs=data)
data_cat.export_data()
```

Where the code issue probably is:
```
hydromt.data_catalog.DataCatalog.export_data()

fn = join(data_root, "data_catalog.yml")
        data_catalog_out.to_yml(fn, root="auto")
```

It seems that adding '../' should fix this.

Cheers

**Comment by dumontgoulart** [comment]
Update: I tried a quick fix that removes the root folder from the path to the locations of the files inside the .yml file. The changes are in bold.

def to_yml(
        self,
        path: Union[str, Path],
        root: str = "auto",
        source_names: List = [],
        used_only: bool = False,
    ) -> None:
        """Write data catalog to yaml format.

        Parameters
        ----------
        path: str, Path
            yaml output path.
        root: str, Path, optional
            Global root for all relative paths in yaml file.
            If "auto" the data source paths are relative to the yaml output ``path``.
        source_names: list, optional
            List of source names to export; ignored if `used_only=True`
        used_only: bool
            If True, export only data entries kept in used_data list, by default False.
        """
        source_names = self._used_data if used_only else source_names
        yml_dir = os.path.dirname(path)
        if root == "auto":
            root = yml_dir
        d = self.to_dict(root=root, source_names=source_names)
        if str(root) == yml_dir:
            d.pop("root", None)  # remove root if it equals the yml_dir
        # Changes below: 
        for k,v in d.items(): 
                for k2, v2 in d[k].items():
                    if k2 == 'path':
                        d[k][k2] = d[k][k2].removeprefix(root+'\\')

          with open(path, "w") as f:
                      yaml.dump(d, f, default_flow_style=False)

**Comment by hboisgon** [comment]
Hi @dumontgoulart. Thanks for spotting this and for you suggestion. This behavior happens if you use a relative path instead of absolute path in the data_root argument of export_data. Maybe indeed something unexpected happens with how 'root' argument of to_yml is treated.

**Comment by DirkEilander** [comment]
@dumontgoulart This should be fixed in the latest version of the main branch (will be released as v0.6.1; date tbd).  It was fixed by deriving the absolute path from the user-provide relative path. We now also write the path as the root in the meta section of the yml file. When moving the data together with the yml file to a different folder that line should be removed. Hope this solves the issue. Feel free to reopen if it doesn't.
